app/src/main/java/com/fstyle/structure_android/data/repository/UserRepository.java

```diff
@@ -37,196 +22,101 @@
-                    @Override
-                    public SingleSource<? extends List<User>> apply(
-                            SearchUserResponse searchUserResponse) throws Exception {
-                        return Single.just(searchUserResponse.getItems());
-                    }
-                });
+    public Completable insertListUser(List<User> users) {
+        return mUserLocalDataSource.insertListUser(Preconditions.checkNotNull(users));
     }
 
     @Override
-    public Completable insertUser(@NonNull final User user) {
-        mDatabase.beginTransaction();
-        return Completable.create(new CompletableOnSubscribe() {
-            @Override
-            public void subscribe(CompletableEmitter emitter) throws Exception {
-                try {
-                    ContentValues values = new ContentValues();
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
-                            user.getSubscriptionsUrl());
-                    mDatabase.insert(UserDbHelper.UserEntry.TABLE_NAME, null, values);
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable insertUser(@NonNull User user) {
+        return mUserLocalDataSource.insertUser(Preconditions.checkNotNull(user));
     }
 
     @Override
-    public Completable updateUser(@NonNull final User user) {
-        mDatabase.beginTransaction();
-        return Completable.create(new CompletableOnSubscribe() {
-            @Override
-            public void subscribe(CompletableEmitter emitter) throws Exception {
-                try {
-                    ContentValues values = new ContentValues();
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
-                            user.getSubscriptionsUrl());
-                    mDatabase.update(UserDbHelper.UserEntry.TABLE_NAME, values,
-                            UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN + "= ?",
-                            new String[] { user.getLogin() });
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable updateUser(@NonNull User user) {
+        return mUserLocalDataSource.updateUser(Preconditions.checkNotNull(user));
```

Comment:
WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Preconditions.checkNotNull' due to existing static import 'com.google.common.base.Preconditions.checkNotNull'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName